pkg/types/validation/installconfig.go

```diff
@@ -456,17 +456,24 @@ func validatePlatform(platform *types.Platform, fldPath *field.Path, network *ty
 	return allErrs
 }
 
-func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
+func validateProxy(c *types.InstallConfig, fldPath *field.Path) field.ErrorList {
```

Comment:
Pass in networking as a separate field, to enforce the concept that the proxy is being validated and not the entire install config.
```suggestion
func validateProxy(p *types.Proxy, networking *types.Networking, fldPath *field.Path) field.ErrorList {
```